File: solutions/move-zeroes/move-zeroes.cpp
// @Title: Move Zeroes
// @Author: colinjxc
// @Date: 2022-01-27T13:10:15+08:00
// @URL: https://leetcode-cn.com/problems/move-zeroes


class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int k = 0;
        for (int j = 0; j < nums.size(); j++) {
            if (nums[j] != 0) {
                if (j != k) {
                    std::swap(nums[k], nums[j]);   
                }
                k++;
            }
        }
    }
};

File: solutions/move-zeroes/move-zeroes.go
// @Title: Move Zeroes
// @Author: colinjxc
// @Date: 2021-05-23T00:12:30+08:00
// @URL: https://leetcode-cn.com/problems/move-zeroes


func moveZeroes(nums []int) {
	k := 0
	for i := range nums {
		if nums[i] != 0 {
			if i != k {
				nums[i], nums[k] = nums[k], nums[i]
			}
			k++
		}
	}
}
